### PR TITLE
Minor Improvements

### PR DESCRIPTION
**Pull request by arturobernalg**
* Add final to variable
* Remove Unused import
* Use Standard Charset object
* Use computeIfAbsent
* Replace 'StringBuffer' with  'StringBuilder'

### CODE REVIEW

File: src/main/java/org/apache/commons/configuration2/io/FileUtils.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
     /**
      * The UTF-8 character set, used to decode octets in URLs.
      */
-    private static final Charset UTF8 = Charset.forName("UTF-8");
+    private static final Charset UTF8 = StandardCharsets.UTF_8;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/io/FileUtils.java:37]: You can now inline this private var.

File: src/main/java/org/apache/commons/configuration2/spring/ConfigurationPropertiesFactoryBean.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
     /** Spring resources for loading configurations **/
     private Resource[] locations;
 
-    /** @see org.apache.commons.configuration2.AbstractConfiguration#throwExceptionOnMissing **/
+    /** @link org.apache.commons.configuration2.AbstractConfiguration#throwExceptionOnMissing **/
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/spring/ConfigurationPropertiesFactoryBean.java:59]: -1 breaks Javadoc, leave as is please.

File: src/main/java/org/apache/commons/configuration2/tree/ModelTransaction.java [hunk 3]
```diff
@@ -361,12 +361,7 @@ Operations fetchOperations(final ImmutableNode target, final int level)
                 Integer.valueOf(level == LEVEL_UNKNOWN ? level(target)
                         : level);
         Map<ImmutableNode, Operations> levelOperations =
-                operations.get(nodeLevel);
-        if (levelOperations == null)
-        {
-            levelOperations = new HashMap<>();
-            operations.put(nodeLevel, levelOperations);
-        }
+                operations.computeIfAbsent(nodeLevel, k -> new HashMap<>());
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/tree/ModelTransaction.java:364]: Fix formatting.

File: src/main/java/org/apache/commons/configuration2/web/AppletConfiguration.java [hunk 4]
```diff
@@ -31,7 +31,7 @@
 public class AppletConfiguration extends BaseWebConfiguration
 {
     /** Stores the wrapped applet.*/
-    protected Applet applet;
+    protected final Applet applet;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/web/AppletConfiguration.java:34]: **Please do NOT break binary compatibility.**

File: src/main/java/org/apache/commons/configuration2/web/ServletConfiguration.java [hunk 5]
```diff
@@ -34,7 +34,7 @@
 public class ServletConfiguration extends BaseWebConfiguration
 {
     /** Stores a reference to the wrapped {@code ServletConfig}.*/
-    protected ServletConfig config;
+    protected final ServletConfig config;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/web/ServletConfiguration.java:37]: **Please do NOT break binary compatibility.**

File: src/main/java/org/apache/commons/configuration2/web/ServletContextConfiguration.java [hunk 6]
```diff
@@ -34,7 +34,7 @@
 public class ServletContextConfiguration extends BaseWebConfiguration
 {
     /** Stores the wrapped servlet context.*/
-    protected ServletContext context;
+    protected final ServletContext context;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/web/ServletContextConfiguration.java:37]: **Please do NOT break binary compatibility.**

File: src/main/java/org/apache/commons/configuration2/web/ServletFilterConfiguration.java [hunk 7]
```diff
@@ -33,7 +33,7 @@
 public class ServletFilterConfiguration extends BaseWebConfiguration
 {
     /** Stores the wrapped filter config.*/
-    protected FilterConfig config;
+    protected final FilterConfig config;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/web/ServletFilterConfiguration.java:36]: **Please do NOT break binary compatibility.**

File: src/main/java/org/apache/commons/configuration2/web/ServletRequestConfiguration.java [hunk 8]
```diff
@@ -34,7 +34,7 @@
 public class ServletRequestConfiguration extends BaseWebConfiguration
 {
     /** Stores the wrapped request.*/
-    protected ServletRequest request;
+    protected final ServletRequest request;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/web/ServletRequestConfiguration.java:37]: **Please do NOT break binary compatibility.**

File: src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java [hunk 9]
```diff
@@ -246,13 +246,13 @@ public PropertiesWriterTestImpl(final ListDelimiterHandler handler) throws IOExc
     private static final String CR = System.getProperty("line.separator");
 
     /** The File that we test with */
-    private static String testProperties = ConfigurationAssert.getTestFile("test.properties").getAbsolutePath();
+    private static final String testProperties = ConfigurationAssert.getTestFile("test.properties").getAbsolutePath();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java:249]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java [hunk 10]
```diff
@@ -246,13 +246,13 @@ public PropertiesWriterTestImpl(final ListDelimiterHandler handler) throws IOExc
...
 
-    private static String testBasePath = ConfigurationAssert.TEST_DIR.getAbsolutePath();
+    private static final String testBasePath = ConfigurationAssert.TEST_DIR.getAbsolutePath();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java:251]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java [hunk 11]
```diff
@@ -246,13 +246,13 @@ public PropertiesWriterTestImpl(final ListDelimiterHandler handler) throws IOExc
...
 
-    private static String testBasePath2 = ConfigurationAssert.TEST_DIR.getParentFile().getAbsolutePath();
+    private static final String testBasePath2 = ConfigurationAssert.TEST_DIR.getParentFile().getAbsolutePath();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java:253]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java [hunk 12]
```diff
@@ -246,13 +246,13 @@ public PropertiesWriterTestImpl(final ListDelimiterHandler handler) throws IOExc
...
 
-    private static File testSavePropertiesFile = ConfigurationAssert.getOutFile("testsave.properties");
+    private static final File testSavePropertiesFile = ConfigurationAssert.getOutFile("testsave.properties");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java:255]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java [hunk 13]
```diff
@@ -35,15 +35,15 @@
 public class TestDefaultExpressionEngine
 {
     /** Stores the names of the test nodes representing tables. */
-    private static String[] tables =
+    private static final String[] tables =
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java:38]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java [hunk 14]
```diff
@@ -35,15 +35,15 @@
...
     { "users", "documents"};
 
     /** Stores the types of the test table nodes. */
-    private static String[] tabTypes =
+    private static final String[] tabTypes =
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java:42]: static finals are in UPPER CASE by convention.

File: src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java [hunk 15]
```diff
@@ -35,15 +35,15 @@
...
     { "system", "application"};
 
     /** Test data fields for the node hierarchy. */
-    private static String[][] fields =
+    private static final String[][] fields =
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/tree/TestDefaultExpressionEngine.java:46]: static finals are in UPPER CASE by convention.

**Review by garydgregory — CHANGES_REQUESTED** [review]
This PR needs a lot of work and shows the limit of whatever automated tooling you are using. Binary compatibility is broken in several places and style conventions like static finals in UPPER CASE are not followed. You MUST review your changes one by one before creating a PR. You MUST also check that you are not breaking binary compatibility; which you can do with a rebase on git master and running `mvn clean install japicmp:cmp`. I've also updated in git master the GitHub build to use the Maven default goal which also checks for BC. Thank you. G.

**Comment by arturobernalg** [comment]
Hi @garydgregory 

I agree that I should have checked if the binaries was broken. My mistake by not running `mvn clean install japicmp:cmp`
I've execute  `japicmp:cmp` and i notice that  are several error, including the two of the UPPERCASE that you comment.
What do you think we should do? we solve all the error of CheckStyle in this PR  or do we create a new one?

**Comment by garydgregory** [comment]
I would just fix checkstyle issues in your PR, not everywhere.

**Comment by arturobernalg** [comment]
> I would just fix checkstyle issues in your PR, not everywhere.

Great. Fix it

File: src/main/java/org/apache/commons/configuration2/interpol/ConstantLookup.java [hunk 16]
```diff
@@ -53,7 +53,7 @@
     private static final char FIELD_SEPRATOR = '.';
 
     /** An internally used cache for already retrieved values. */
-    private static Map<String, Object> constantCache = new HashMap<>();
+    private static final Map<String, Object> constantCache = new HashMap<>();
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/configuration2/interpol/ConstantLookup.java:56]: @arturobernalg 
Again, static finals are in UPPER CASE by convention, so `constantCache` becomes `CONSTANT_CACHE`. PLEASE, review the whole PR before submission.

**Comment by arturobernalg** [comment]
HI @garydgregory 
I think everything would be solved by now.
Regards

File: src/test/java/org/apache/commons/configuration2/test/HsqlDB.java [hunk 17]
```diff
@@ -35,7 +35,7 @@
 public class HsqlDB
 {
     private Connection connection = null;
-    private static Log log = LogFactory.getLog(HsqlDB.class);
+    private static final Log log = LogFactory.getLog(HsqlDB.class);
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/configuration2/test/HsqlDB.java:38]: Again, please, please, please, review ALL of your changes. If you make a static final, then the name should become UPPER CASE by convention.

File: src/main/java/org/apache/commons/configuration2/interpol/ConstantLookup.java (hunk 16 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/configuration2/interpol/ConstantLookup.java]: Hi @garydgregory
I have double check and it seems that they are all changed already

**Pull request closed without merging**